test/pytest/test_bridge.py

```diff
@@ -411,6 +411,21 @@ async def test_fsread1_errors(transport):
                                reply_keys={'message': "attribute 'max_read_size': must have type int"})
 
 
+@pytest.mark.asyncio
+async def test_fsread1_size_hint(transport):
+    ch = transport.send_open('fsread1', path='/etc/passwd', binary='raw')
```

Comment:
Please also add a case that opens a channel in text mode and confirms that the hint is absent.

Reply:
Please also add a case that opens a fifo and/or a character device in binary mode and confirms that the hint is absent.  The fifo case would be particularly interesting because you can feed it some non-zero amount of data.